src/ai/backend/common/docker.py

```diff
@@ -258,7 +258,7 @@ async def login(
         return {"auth": basic_auth, "headers": {}}
     elif ping_status == 404:
         raise RuntimeError(f"Unsupported docker registry: {registry_url}! (API v2 not implemented)")
-    elif ping_status == 401:
+    elif ping_status in [400, 401]:
```

Comment:
`400` does not usually come as a response indicating invalid credentials so please briefly write down a comment about why it is essential.

Reply:
For reasons that are unclear, the AWS ECR-public server responds with a 400 return code when there is an issue with the token value. In contrast, the ECR private server responds with a 401 return code.

Please refer to the results of the command below.


```
jopemachine in 🌐 ubuntu in ~/backend.ai🔒 on ☁️  (us-east-1)
❯ curl -I -k -H "Authorization: Bearer abc" https://public.ecr.aws/v2
HTTP/2 400
date: Sat, 03 Aug 2024 06:00:56 GMT
content-type: application/json; charset=utf-8
content-length: 80
docker-distribution-api-version: registry/2.0


jopemachine in 🌐 ubuntu in ~/backend.ai🔒 on ☁️  (us-east-1)
❯ curl -I -k https://public.ecr.aws/v2
HTTP/2 401
date: Sat, 03 Aug 2024 06:01:01 GMT
content-type: application/json; charset=utf-8
content-length: 58
docker-distribution-api-version: registry/2.0
www-authenticate: Bearer realm="https://public.ecr.aws/token/",service="public.ecr.aws",scope="aws"


jopemachine in 🌐 ubuntu in ~/backend.ai🔒 on ☁️  (us-east-1)
❯ curl -I -k https://<...>.dkr.ecr.us-east-1.amazonaws.com/v2
HTTP/1.1 401 Unauthorized
Docker-Distribution-Api-Version: registry/2.0
Sizes:
Www-Authenticate: Basic realm="https://<...>.dkr.ecr.us-east-1.amazonaws.com/",service="ecr.amazonaws.com"
Date: Sat, 03 Aug 2024 06:01:35 GMT
Content-Length: 15
Content-Type: text/plain; charset=utf-8


jopemachine in 🌐 ubuntu in ~/backend.ai🔒 on ☁️  (us-east-1)
❯ curl -I -k -H "Authorization: Bearer abc" https://<...>.dkr.ecr.us-east-1.amazonaws.com/v2
HTTP/1.1 401 Unauthorized
Docker-Distribution-Api-Version: registry/2.0
Sizes:
Www-Authenticate: Basic realm="https://<...>.dkr.ecr.us-east-1.amazonaws.com/",service="ecr.amazonaws.com"
Date: Sat, 03 Aug 2024 06:01:54 GMT
Content-Length: 15
Content-Type: text/plain; charset=utf-8
```